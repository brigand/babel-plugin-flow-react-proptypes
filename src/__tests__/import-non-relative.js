const babel = require('babel-core');
const content = `
import type {NamedType} from 'foo';
import type DefaultType from 'bar';

type FooProps = {
  an_imported_named_type: NamedType,
  an_imported_default_type: DefaultType,
  a_global_type: Date,
  a_undefined_type: FooBarBaz,
};

class C extends React.Component {
    props: FooProps
};
`;

it('import-object', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
