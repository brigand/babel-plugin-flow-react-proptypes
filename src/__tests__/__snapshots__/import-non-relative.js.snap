// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-object 1`] = `
"'use strict';

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var C = function (_React$Component) {
  _inherits(C, _React$Component);

  function C() {
    _classCallCheck(this, C);

    return _possibleConstructorReturn(this, (C.__proto__ || Object.getPrototypeOf(C)).apply(this, arguments));
  }

  return C;
}(React.Component);

C.propTypes = {
  an_imported_named_type: function an_imported_named_type() {
    return (typeof NamedType === 'function' ? _propTypes2.default.instanceOf(NamedType).isRequired : _propTypes2.default.any.isRequired).apply(this, arguments);
  },
  an_imported_named_namespace_type: function an_imported_named_namespace_type() {
    return (typeof NamespaceType === 'function' ? _propTypes2.default.instanceOf(NamespaceType).isRequired : _propTypes2.default.any.isRequired).apply(this, arguments);
  },
  an_imported_default_type: function an_imported_default_type() {
    return (typeof DefaultType === 'function' ? _propTypes2.default.instanceOf(DefaultType).isRequired : _propTypes2.default.any.isRequired).apply(this, arguments);
  },
  a_global_type: function a_global_type() {
    return (typeof Date === 'function' ? _propTypes2.default.instanceOf(Date).isRequired : _propTypes2.default.any.isRequired).apply(this, arguments);
  },
  a_undefined_type: function a_undefined_type() {
    return (typeof FooBarBaz === 'function' ? _propTypes2.default.instanceOf(FooBarBaz).isRequired : _propTypes2.default.any.isRequired).apply(this, arguments);
  }
};
;"
`;
